### PR TITLE
Setup CI

### PR DESCRIPTION
**Pull request by darwin67**
Setup CI for basic tests and linting.
Also fix linter errors, and added dependabot for auto update with security vulnerabilities.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -3,8 +3,15 @@ root = true
...
+indent_size = 2
 indent_style = space
-insert_final_newline = false
+insert_final_newline = true
 max_line_length = 120
+tab_width = 2
+
+[*.{kt,kts}]
+indent_size = 4
 tab_width = 4
```
Comment (djfarrelly) [inline .editorconfig:14]: 👍  We may also want to add .java files here as well as there is likely to be some Java files in this repo :)

File: Makefile [hunk 2]
```diff
@@ -5,3 +5,11 @@ dev:
...
 test:
 	gradle test inngest-core:test
+
+.PHONY: lint
+lint:
+	ktlint --color
+
+.PHONY: fmt
+fmt:
+	ktlint -F
```
Comment (djfarrelly) [inline Makefile:15]: Nice! I didn't know about this!

**Review by djfarrelly — COMMENTED** [review]
Great stuff!

File: .editorconfig (hunk 1 above)
Comment (darwin67) [reply .editorconfig:14]: added here. https://github.com/inngest/inngest-kt/pull/13

**Pull request merged**